File: develnext-designer/src/main/java/org/develnext/jphp/gui/designer/classes/UXFormObjectTreeValue.java
package org.develnext.jphp.gui.designer.classes;

import javafx.scene.Node;
import org.develnext.jphp.gui.designer.GuiDesignerExtension;
import org.develnext.jphp.gui.designer.editor.tree.FormObjectTreeValue;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.lang.BaseWrapper;
import php.runtime.reflection.ClassEntity;

@Reflection.Namespace(GuiDesignerExtension.NS)
public class UXFormObjectTreeValue extends BaseWrapper<FormObjectTreeValue> {
    interface WrappedInterface {
        @Property String id();
        @Property String typeName();
        @Property Node icon();
        @Property boolean renameable();
    }

    public UXFormObjectTreeValue(Environment env, FormObjectTreeValue wrappedObject) {
        super(env, wrappedObject);
    }

    public UXFormObjectTreeValue(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }
}

File: develnext-designer/src/main/resources/JPHP-INF/sdk/php/gui/designer/UXFormObjectTreeValue.php
<?php
namespace php\gui\designer;

use php\gui\UXNode;

class UXFormObjectTreeValue
{
    /**
     * @var string
     */
    public $id;

    /**
     * @var string
     */
    public $typeName;

    /**
     * @var UXNode
     */
    public $icon;

    /**
     * @var bool
     */
    public $renameable = true;
}
